src/DateTimePicker.jsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import cx from 'classnames';
+import Flatpickr from './FlatpickrComponent';
+import TimeInput from './TimeInput';
+import DateTimeLocalInput from './DateTimeLocalInput';
+
+/**
+ * @module DateTimePicker
+ */
+class DateTimePicker extends React.Component {
+
+	constructor(props) {
+		super(props);
+
+		const propsValue = props.value || {};
+		this.state = {
+			value: {
+				date: propsValue.date || '',
+				time: propsValue.time || ''
+			},
+			isDateTimeLocalSupported: false
+		};
+
+		this.setDate = this.setDate.bind(this);
+		this.setTime = this.setTime.bind(this);
+		this.parseDateTime = this.parseDateTime.bind(this);
+	}
+
+	componentWillMount() {
+		this.setState({ isDateTimeLocalSupported: this.hasBrowserSupport() });
+	}
+
+	hasBrowserSupport() {
+		const input = document.createElement('input'),
+			invalidValue = 'notadate';
+
+		// some browsers (as Android stock browsers) pretend they support
+		// certain input types,
+		// so we set the value to an invalid value and see if browser rejects
+
+		input.setAttribute('type', 'datetime-local');
+		input.setAttribute('value', invalidValue);
+
+		return !(this.props.forceFlatpickr || input.value === invalidValue);
+	}
+
+	setDate(value) {
+		this.setState({ value: { date: value }});
+	}
+
+	setTime(value) {
+		this.setState({ value: { time: value }});
+	}
+
+	parseDateTime(value) {
+		const datetime = value.split('T'),
+			newState = {
+				date: datetime[0],
+				time: datetime[1]
+			};
+		this.setState(newState);
+	}
+
+	getDateTime() { return `${this.state.value.date}T${this.state.value.time}`; }
+
+	render() {
+		const {
+			callback,			// eslint-disable-line no-unused-vars
+			className,
+			dateOnly,
+			datepickerOptions,  // eslint-disable-line no-unused-vars
+			id,
+			label,
+			value,				// eslint-disable-line no-unused-vars
+			required,
+			...other
+		} = this.props;
+
+		const classNames = cx(
+			'dateTimePicker',
+			{required},
+			className
+		);
+
+		const labelClassNames = cx({required});
+		if (this.state.isDateTimeLocalSupported) {
+
+			const dateTimeLocalOpts = {
+				min: `${datepickerOptions.minDate}T00:00:00`,
+				max: `${datepickerOptions.maxDate}T00:00:00`
```

Comment:
check out `new Date(datepickerOptions.maxDate).toIsoString()` - it includes UTC timezone info so it might not work as expected out of the box, but it avoids having to manually construct an ISO string